Control flow exercise

Tejpbit opened the pull request:
Created very basic control flow exercise.
First explains how loop and if statements work.
Continue by showing a fizzbuzz program with the if-statements in the wrong order.
Then a loosely implemented prime number exercise where there's some logic missing in the is_prime() function.

I'm thinking this is an exercise for those who've never seen programming before. Maybe this fills the same place as codecadamy would? But it's an exercise to get them started quickly and let them complete an exercise.

This is more in the spirit which we talket about, (let them have some code to work with instead of a blank slate).

Feel free to comment and discuss improvements!

niklasbroberg commented:
I think this is a great start, exactly what I'm looking for! I have some small suggestions for improvements, listed below, but overall this is really great.

This exercise, and others like it, would indeed fill "the same place codeacademy would". But I think there's a point in having our own exercises and not send the students offsite right away. I think it's important that we foster the feeling that we are prepared for the beginners, and that they feel welcome. Having exercises like this will be significant for that.

I think you do a great job here introducing ifs and loops. I also think you tie it all together really nicely at the end. The introduction of functions, and even more so the introduction of returns, is not active though, you're just describing them in the readme file. Could you envision having some simple exercise that lets them practice functions and returns, before the final isprime example?

Tejpbit commented:
I've now added exercises to introduce the student to lists which is then used to introduce the student to functions and returns.
Added additional extension on the last prime exercise to create a function there aswell.
